datalad/support/sshconnector.py

```diff
@@ -450,7 +314,351 @@ def get_git_version(self):
 
 
 @auto_repr
-class SSHManager(object):
+class SingleProcessSSHConnection(BaseSSHConnection):
+    """Representation of an SSH connection.
+
+    The connection is opened for execution of a single process, and closed
+    as soon as the process end.
+    """
+    def __init__(self, sshri, **kwargs):
+        """Create a connection handler
+
+        The actual opening of the connection is performed on-demand.
+
+        Parameters
+        ----------
+        sshri: SSHRI
+          SSH resource identifier (contains all connection-relevant info),
+          or another resource identifier that can be converted into an SSHRI.
+        **kwargs
+          Pass on to BaseSSHConnection
+        """
+        super().__init__(sshri, **kwargs)
+        self._ssh_open_args += [
+            # we presently do not support any interactive authentication
+            # at the time of process execution
+            '-o', 'PasswordAuthentication=no',
+            '-o', 'KbdInteractiveAuthentication=no',
```

Comment:
hm, I know that it is "inefficient", "annoying", and might be just infeasible in proper real-life cases etc but why to disable completely ?  Especially for `maint`... I wonder if may be worth adding a dedicated config option, e.g. `datalad.ssh.allow-password`.  We could even set it to False right away since it would be hit only on Windows (by default)... That still would allow where/when really needed for password based authentication, even though demanding serial execution, multiple entries, etc

Reply:
We have discussed that in the call this week. It is not possible for most calls, because stdout/err is consumed by some kind of handler. I am sure somehow this could be made possible. But I don't know how, and I have no plans to implement it, in this context of this urgent fix.

Reply:
ok.